arm/Microsoft.Insights/scheduledQueryRules/deploy.bicep

```diff
@@ -30,148 +45,56 @@ param workspaceResourceId string
-@allowed([
-  'AlertingAction'
-  'LogToMetricAction'
-])
-param odataType string = 'AlertingAction'
+@description('Optional. The rule criteria that defines the conditions of the scheduled query rule.')
+param criterias object = {}
 
-@description('Optional. Suppress Alert for (in minutes).')
-param suppressForMinutes int = 0
+@description('Optional. Mute actions for the chosen period of time (in ISO 8601 duration format) after the alert is fired. If set, autoMitigate must be disabled.Relevant only for rules of the kind LogAlert.')
+param suppressForMinutes string = ''
 
 @description('Optional. Tags of the resource.')
 param tags object = {}
 
 @description('Optional. Customer Usage Attribution id (GUID). This GUID must be previously registered')
 param cuaId string = ''
 
-var metricTrigger = {
-  metricColumn: metricColumn
-  metricTriggerType: breachesTriggerType
-  threshold: breachesThreshold
-  thresholdOperator: breachesThresholdOperator
-}
-
 module pid_cuaId '.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
   name: 'pid-${cuaId}'
   params: {}
 }
 
-resource queryAlert 'microsoft.insights/scheduledQueryRules@2018-04-16' = {
+resource queryAlert 'Microsoft.Insights/scheduledQueryRules@2021-02-01-preview' = {
   name: alertName
   location: location
   tags: tags
+  kind: kind
   properties: {
+    actions: {
+      actionGroups: actions
+      customProperties: {}
+    }
+    autoMitigate: contains(kind, 'LogAlert') ? autoMitigate : null
+    criteria: criterias
+
     description: alertDescription
+    displayName: alertName
     enabled: enabled
-    source: {
-      query: query
-      authorizedResources: authorizedResources
-      dataSourceId: workspaceResourceId
-      queryType: 'ResultCount'
-    }
-    schedule: {
-      frequencyInMinutes: evaluationFrequency
-      timeWindowInMinutes: windowSize
-    }
-    action: {
-      severity: severity
-      aznsAction: {
-        actionGroup: actions
-      }
-      throttlingInMin: suppressForMinutes
-      trigger: {
-        thresholdOperator: metricResultCountThresholdOperator
-        threshold: metricResultCountThreshold
-        metricTrigger: (empty(metricColumn) ? null : metricTrigger)
-      }
-      criteria: criterias
-      'odata.type': 'Microsoft.WindowsAzure.Management.Monitoring.Alerts.Models.Microsoft.AppInsights.Nexus.DataContracts.Resources.ScheduledQueryRules.${odataType}'
-    }
+    evaluationFrequency: contains(kind, 'LogAlert') ? evaluationFrequency : null
+    muteActionsDuration: contains(kind, 'LogAlert') ? suppressForMinutes : null
+    overrideQueryTimeRange: contains(kind, 'LogAlert') ? queryTimeRange : null
+    scopes: scopes
+    severity: contains(kind, 'LogAlert') ? severity : null
+    skipQueryValidation: contains(kind, 'LogAlert') ? skipQueryValidation : null
+    targetResourceTypes: contains(kind, 'LogAlert') ? targetResourceTypes : null
+    windowSize: contains(kind, 'LogAlert') ? windowSize : null
   }
 }
```

Comment:
Please add `@description('...')` to the outputs and regenerate the readme

Reply:
Done :)